include/net/net_pkt.h

```diff
@@ -178,6 +178,13 @@ struct net_pkt {
 					     * AF_UNSPEC.
 					     */
 		u8_t ppp_msg           : 1; /* This is a PPP message */
+
+		u8_t tcp_first_msg     : 1; /* Is this the first time this
```

Comment:
I'd find both this code comment and commit message which introduces it ("net: pkt: Add flag for telling if this is TCP 1st packet") to be unclear and confusing. To start with, we already maintain linked list of packets which were already sent (and may need to be resent) as `tcp->sent_list`. So, if a packet is not in that list, it's "1st packet" (ahem to the terminology), and if it's in that list, it's "not 1st packet". So, do we need an extra flag for that? If we do, how about naming it more clear? E.g., "resent" (or "resent_pkt", or "pkt_resent"). But note that we already have flags `sent_or_eof` and `pkt_queued`. So, as I said, this is getting more and more confusing.

Reply:
The flag tells is this the 1st time **this** packet has been sent. This is done in order to avoid doing one extra ref when sending it first time. It is indeed confusing, I try to clarify the documentation a bit. The `sent_or_eof` flag is for another purpose, there it means whether the pkt was sent in the first place. I tried to use the sent flag for detecting whether we need to ref the pkt in relevant places but it became very convoluted so it was much easier to just add a new flag for this. I agree that there are several flags now in the system and their interaction is hairy, but it is not easy to send the packet as there are two threads involved and the interaction is difficult when we have timers that expire when the packet is not even sent yet.

Reply:
>  I agree that there are several flags now in the system and their interaction is hairy, but it is not easy to send the packet as there are two threads ...

Ok, thanks for clarification.